inc/saitunnel.h

```diff
@@ -957,7 +957,6 @@ typedef enum _sai_tunnel_term_table_entry_attr_t
      *
      * @type sai_ip_address_t
      * @flags MANDATORY_ON_CREATE | CREATE_ONLY
-     * @condition SAI_TUNNEL_TERM_TABLE_ENTRY_ATTR_TYPE == SAI_TUNNEL_TERM_TABLE_ENTRY_TYPE_P2P or SAI_TUNNEL_TERM_TABLE_ENTRY_ATTR_TYPE == SAI_TUNNEL_TERM_TABLE_ENTRY_TYPE_P2MP
      */
```

Comment:
Not clear why the condition check is removed

Reply:
When condition is met, then the attribute is mandatory during create, if condition is not met, then attribute is not valid on create and meta checker will not allow it to be passed. However we do want the attribute SAI_TUNNEL_TERM_TABLE_ENTRY_ATTR_DST_IP for new tunnel types MP2P and MP2MP as well.

Reply:
Instead of removing the condition, would it make sense to add new conditions (even if its the complete set of enums)

Reply:
I don't think so. It's semantically the same, and less confusing to understand "this is required for all tunnel types" than "this is required for P2P AND P2MP AND MP2P AND MP2MP"

Reply:
I agree that semantically it's the same but to keep backward compatibility more obvious adding new condition checks seems more reasonable then completely removing the condition check. It's also quite possible that there may be some other attribute in future that MAY need the condition check. So what looks like a full set right now may not be so in future.

Reply:
@JaiOCP Are other existing attributes designed for a possibility of new tunnel types? The way I see it, right now this is broken. We cannot use MP2P and MP2MP tunnels. We should fix the case for existing tunnel types. Can we please come to a conclusion on this? We could meet if required.

Reply:
My preference is to add the new checks instead of removing the condition flag completely.
Rational is that adding new checks makes the intent clear vs removing the condition check can be interpreted as if no condition is check is needed or is needed for all the options i.e. intent is ambiguous.

Reply:
@kcudnik Do you agree with this? If so, we'll go ahead and make the change

Reply:
@JaiOCP do you mean add new check like adding condition for all existing enums then ?

Reply:
> @JaiOCP do you mean add new check like adding condition for all existing enums then ?

yes. If we add new enums in future then we need to again add the condition flag.
Also removing condition flag does tooling treat as backward incompatible change ?

Reply:
i think that was the point when i talked with google, that we want to remove condition at all, so even if we add new enums in the future this attribute will be required for them too

Reply:
Then as I mentioned my preference is to keep the condition flag.

Reply:
fine with me

Reply:
@kcudnik @JaiOCP could you please take a look now?